src/pfe/portal/routes/templates.route.js

```diff
@@ -59,17 +57,22 @@ router.post('/api/v1/templates/repositories', validateReq, (req, res, _next) =>
     res.status(400).send("Invalid repository URL");
     return;
   }
-  if (!user.templates.addRepository(repositoryUrl, repositoryDescription)) {
-    res.status(400).send("Repository url already available.");
-    return;
+  try {
+    await user.templates.addRepository(repositoryUrl, repositoryDescription);
+  } catch (error) {
+    if (error.message === 'Repository URL must be unique') {
```

Comment:
this should really be in a constant in the callee file so that we aren't checking against a string that is declared in a totally separate place, in case it changes